### ISSUE TITLE
TriggerHandler

### ISSUE THREAD

**Issue by alfalfa6945** [Courseplay/courseplay#6099]
Thank you all for the efforts in making courseplay! A slight issue with the TriggerHandler.lua in the latest master (6.02.00063):

1) What value does "trigger" have when accessed from line 935?
2) What value does "spec" have in the overwriten function "setFillUnitIsFilling()"

Also, is there an existing issue of a value not resetting when courseplay refills? As an example, when spreading slurry (default Giant's vehicles), using 2 courses (first is the spreading course, second is the filling course), when first empty, courseplay will drive the refill course and refill and then continue the spreading course. However, when empty the second time, it will drive the refill course and simply wait at the fillTrigger without actually refilling from it (refill trigger is a Kotte Garant FRC).

**Comment by Tensuko** [comment]
logfile with debugchannel 2.

@schwiti6190 sounds a bit like #6093

**Comment by alfalfa6945** [comment]
Hi Tensuko, I don't think I need to trouble you with a logfile as I believe schwiti6190 has fixed the issues I have described in the latest (6.02.00062) "David" branch.
In that version, "spec" has been given a value in "setFillUnitIsFilling()" and the updateFillUnitTriggers() function has been removed and therefore the issue with the "trigger" value is gone.

As an aside, with the latest "David" branch/build, there is one issue on line 840 that will prevent it from functioning. If "lastCallbackData" is nil, then there will be an error indexing "lastCallbackData.callback" in the "or" portion of the code. Once fixed, the problem with regards to spreading/filling slurry I had described has been cured.

Thank you all very much!
